autogpt/api/v1/endpoints/agents.py

```diff
@@ -0,0 +1,31 @@
+import uuid
+
+from fastapi import APIRouter
+from pydantic.main import BaseModel
+
+router = APIRouter()
+
+
+class WorkspaceConfigRequest(BaseModel):
+    root: str
+
+
+class WorkspaceRequest(BaseModel):
+    configuration: WorkspaceConfigRequest
+
+
+class AgnosticAgentRequest(BaseModel):
+    workspace: WorkspaceRequest
+
+
+class AgentRequest(AgnosticAgentRequest):
+    pass
```

Comment:
The Request for the agnostic agent will be different than the AgentRequest, which will be Auto-GPT specific